api/v1alpha1/extension_types.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Although this looks like a bool, there might
+type ExtensionManaged string
```

Comment:
More of a nit, but this name feels like it could be a bit misleading without looking more into the type when writing some code with it. Maybe something like:
- `ExtensionManagedState`
- `ManagedState` (I prefer this as it is shorter but less "descriptive" so I could see an argument for the longer one)